protocols/noise/src/keys.rs

```diff
@@ -18,156 +18,185 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use crate::NoiseError;
-use curve25519_dalek::{
-    constants::{X25519_BASEPOINT, ED25519_BASEPOINT_POINT},
-    edwards::CompressedEdwardsY,
-    montgomery::MontgomeryPoint,
-    scalar::Scalar
-};
-
-#[derive(Debug, Clone)]
-pub enum Curve25519 {}
-
-#[derive(Debug, Clone)]
-pub enum Ed25519 {}
-
-/// ECC public key.
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct PublicKey<T> {
-    bytes: [u8; 32],
-    _marker: std::marker::PhantomData<T>
+use curve25519_dalek::edwards::CompressedEdwardsY;
+use libp2p_core::identity::ed25519;
+use ring::digest::{SHA512, digest};
+use x25519_dalek as x25519;
+
+/// X25519 static keypair.
+pub struct StaticKeypair {
+    secret: x25519::StaticSecret,
```

Comment:
We might abstract Noise over the curve, perhaps as part of trying to get https://noiseexplorer.com/ to generate Rust, but I'd maybe first ask Trevor and/or search the [Noise mailing list archives](https://moderncrypto.org/mail-archive/noise/) for any discussion about this. 

Aside from Cruve448, I'd think the target curves might be FourQ because it's possibly faster and only very slightly less secure, and the Ristretto group for when you want complex zk protocols integrated with your key exchange.  I'd avoid secp256k1 here myself.

Reply:
Curve 448 should be uncontroversial as it is already part of [the spec](http://noiseprotocol.org/noise.html#the-448-dh-functions).

Reply:
I will ty to come up with some changes to keep the door open for other curves in the future.

Reply:
I create a new PR [here](https://github.com/romanb/rust-libp2p/pull/1) against the branch of this PR that shows the approach I came up with so far to generalise over the choice of DH.